Workiva/over_react
# CPLAT-13170 Analyze over_react examples without generated code

## Motivation
  
In order to verify that there are no edge cases we missed in the new boilerplate, we need to run analysis in CI pre-build without the generated files checked in. Because pre-build analysis without generated files won't run cleanly in Dart 2.9 (because we can't ignore `undefined_identifier` errors), we can't run pre-build analysis on the whole repo so we will just run it on examples of each boilerplate.

## Changes
  
- 22ced1c / d1fbdd6 Stop checking in generated files in tests/examples where possible by updating `build.yaml` to build to cache.
- 37d3493 Add missing ignore comments to tests/examples/web so that they will analyze cleanly without a build.
- 2509241 Add examples of each boilerplate to the `example` directory.
- 8d20eb9 Update GitHub actions to run pre-build analysis on the examples in Dart 2.7.2 and Dart 2.12.x (skipping Dart stable because this will fail on Dart >=2.9 and < 2.12)
  - As a part of this, update CI analysis check to run `tuneup check` for Dart 2.7 and `dart analyze` for Dart >2.10 because `dartanalyzer` doesn't use the analysis server and won't see the generated files (see discussion [here](https://workiva.slack.com/archives/GFELW7RRB/p1611858119009300))

#### Release Notes
  

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
        
      - [ ] Verify CI passes (for Dart 2.7.2 and Dart stable)
        - for Dart 2.7.2 CI, verify that:
          - [ ] the pre-build analysis step runs `tuneup check` on just `example/boilerplate_versions`
          - [ ] the post-build analysis step runs `tuneup check` on the whole project
        - for Dart stable CI, verify that:
          - [ ] the pre-build analysis step does not run analysis
          - [ ] the post-build analysis step runs `dart analyze` on the whole project
        - for Dart dev CI, verify that:
          - [ ] the pre-build analysis step runs `dart analyze` on just `example/boilerplate_versions/mixin_based/dart_2_9_compatible`
          - [ ] the pre-build analysis step passes
      - [ ] Serve `example` and verify that the boilerplate version testing page doesn't have any errors

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:


[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria
